src/api/cvc4cpp.cpp

```diff
@@ -2239,155 +2255,155 @@ Op Term::getOp() const
   // Notice this is the only case where getKindHelper is used, since the
   // cases above do not have special cases for intToExtKind.
   return Op(d_solver, getKindHelper());
+  ////////
+  CVC4_API_TRY_CATCH_END;
 }
 
-bool Term::isNull() const { return isNullHelper(); }
+bool Term::isNull() const
+{
+  CVC4_API_TRY_CATCH_BEGIN;
+  //////// all checks before this line
+  return isNullHelper();
+  ////////
+  CVC4_API_TRY_CATCH_END;
+}
 
 Term Term::getConstArrayBase() const
 {
   NodeManagerScope scope(d_solver->getNodeManager());
+  CVC4_API_TRY_CATCH_BEGIN;
```

Comment:
Is the NodeManagerScope above this line necessary?

Reply:
I don't know but I'm tempted to assume it is. Ideally all of the functions should open a scope, but as discussed online, that is a huge can of worms that needs to be fixed.

Reply:
Ah I see, I actually though you were opening as part of one of these macros.  I guess it's fine to keep for now.

Reply:
Yeah, adding this to the macro is the next step, but my initial attempt failed, so I'm doing this incrementally. We're close, though, now, so I'll try to add it after the refactor is in.